### PR TITLE
Clarify behavior of `\tl_if_novalue:n(TF)`

### PR DESCRIPTION
**Pull request by muzimuzhi**
See #1402.

### CODE REVIEW

File: l3kernel/l3tl.dtx [hunk 1]
```diff
@@ -330,8 +330,12 @@
...
 %     \cs{tl_if_novalue:nTF} \Arg{token list} \Arg{true code} \Arg{false code}
 %   \end{syntax}
-%   Tests if the \meta{token list} is exactly equal to the special
-%   \cs{c_novalue_tl} marker. This function is intended to allow construction
+%   Tests if the \meta{token list} is exactly equal to the content of special
+%   \cs{c_novalue_tl} marker, hence it is functionally the same as
+%   \begin{quote}
+%     \cs{tl_if_eq:NnTF} \cs{c_novalue_tl} \Arg{token list} \Arg{true code} \Arg{false code}
+%   \end{quote}
+%   but expandable. This function is intended to allow construction
```
Comment (FrankMittelbach) [inline l3kernel/l3tl.dtx:338]: sorry, but I find my explanation easier to understand. Yours is certainly a correct description but I think it is hard from your explanation to deduce that \tl_if_novalue:nTF \c_novalue_tl returns false.

But yes my explanation could most likely be improved in the wording.

Comment (car222222) [reply l3kernel/l3tl.dtx:338]: The longer version here makes more sense to me (with minor enhancements:-):

`Tests whether the \meta{token list} is exactly equal to the content of the special  \cs{c_novalue_tl} marker token list.`

Not sure about the extra explanations, maybe both forms are needed?

Comment (muzimuzhi) [reply l3kernel/l3tl.dtx]: @FrankMittelbach I replaced my `\tl_if_eq:NnTF \c_novalue_tl {<token list>}` with something adapted from your explanation in https://github.com/latex3/latex3/issues/1402#issuecomment-1873461713. The fact that `\c_novalue_tl` is not a quark is told can be seen in `\c_novalue_tl`'s doc, so I dropped that mention to quark.

@car222222 "Tests if ... " is used in documentation of all other `\tl_if_...` conditionals, so I decide to stick to this phrase. `\c_novalue_tl`'s name already tells it's a token list variable, thus I hesitate to add an extra "token list" tail.

"both in respect of character codes and category codes" is added, to be aligned with docs of `\tl_if_equal:(NN|Nn|nn)(TF)` functions.

Comment (car222222) [reply l3kernel/l3tl.dtx]: > Tests if ... " is used in documentation of all 

Sure! (but I would change all of them:-)

**Review by car222222 — APPROVED** [review]
Latest version looks very good to me. Thanks!

**Review by FrankMittelbach — APPROVED** [review]

**Pull request merged**